### PR TITLE
Create Opera device which sets Opera specific capabilities

### PR DESCRIPTION
**Pull request by emakiej**
The purpose of these changes is to make it as easy as possible to setup typical tests using Opera browser.
Also they let the user to use opera analogously to other browsers supported by Appium.

### CODE REVIEW

**Comment by jlipps** [comment]
Sure, we'll review it when we get to 1.4! Thanks for this.

**Comment by 0x1mason** [comment]
@jlipps You want to merge this?

File: lib/devices/android/opera.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  this.init(desiredCaps);
+};
+
+Opera.prototype.init = function (desiredCaps) {
+  if (desiredCaps.automationName && desiredCaps.automationName.indexOf('selendroid') !== -1) {
+    _.defaults(Opera.prototype, Selendroid.prototype);
+    Opera.prototype._installAppForTest = Selendroid.prototype.installAppForTest;
+    Opera.prototype._init = Selendroid.prototype.init;
+  }
+  else {
```
Comment (jlipps) [inline lib/devices/android/opera.js]: `} else {`

File: lib/devices/android/opera.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  this.avoidProxy = _.union(this.avoidProxy, [
+                     ['POST', new RegExp('^/wd/hub/session/[^/]+/touch/perform')],
+                     ['POST', new RegExp('^/wd/hub/session/[^/]+/touch/multi/perform')]
+                   ]);
+};
+
+var operaBase = {};
+
+operaBase.installAppForTest = function (cb) {
+	this
```
Comment (jlipps) [inline lib/devices/android/opera.js]: should be 2 spaces per indent in this method

File: lib/devices/android/opera.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			.shell(
+					"echo 'opera --disable-fre --enable-remote-debugging --metrics-recording-only' > /data/local/tmp/" +
+					deducedCapabilities.commandLineFile,
+					cb);
+		}
+	}.bind(this));
+};
+
+operaBase.defaultWebviewName = function () {
+	return "CHROMIUM";
```
Comment (jlipps) [inline lib/devices/android/opera.js]: indent

File: lib/devices/android/opera.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+function getDeducedCapabilities(packageName) {
+  var productName = "", betaPart = "";
+  var packageParts = packageName.split(".");
+  for (var i = 0; i < packageParts.length; i++) {
+    var part = packageParts[i];
+    if (part === "opera" || part === "mini") {
+      productName = part;
+    }
```
Comment (jlipps) [inline lib/devices/android/opera.js]: `} else if {`

**Comment by jlipps** [comment]
I like this change; made a few comments

**Comment by Jonahss** [comment]
@emakiej are you able to make the adjustments that @jlipps mentioned above?

**Comment by emakiej** [comment]
Hi @Jonahss , @jlipps .
I'm having a busy time now. I'll be able to follow up the review next weekend (18-19 IV). Isn't that to late for you?

**Comment by Jonahss** [comment]
@emakiej  ok :)  just making sure things weren't dead.

**Comment by jlipps** [comment]
hi @emakiej do you have any time to make the changes proposed here?

**Comment by jlipps** [comment]
closing for now, feel free to reopen with the changes. thanks!

**Pull request closed without merging**